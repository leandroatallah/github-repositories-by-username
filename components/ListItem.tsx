import React from "react"
import styled from "styled-components"
import moment from "moment"
import Image from "next/image"

const ListItemContainer = styled.div`
  padding: 20px;
  border-radius: 5px;
  margin-bottom: 20px;
  cursor: pointer;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  background-color: rgba(255, 255, 255, 0.02);
  transition: all 0.15s ease;

  &:hover {
    transform: translateY(-2px);
  }
`
const ListItemContainerFooter = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 20px;
  font-size: 13px;
  font-weight: 400;

  @media (max-width: 748px) {
    flex-direction: column;
    align-items: flex-start;
  }
`

const FooterCounts = styled.div`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;

  @media (max-width: 748px) {
    margin-top: 12px;
  }
`

const Title = styled.h2`
  font-size: 20px;
  font-weight: bold;
  margin: 0;
  color: #aaf;

  @media (max-width: 748px) {
    margin-bottom: 24px;
  }
`

const CountICon = styled.div`
  display: inline-flex;
  align-items: center;
  margin-left: 10px;
  font-size: 0.9rem;

  @media (max-width: 748px) {
    margin-left: 0;
  }

  div {
    display: inline-block;
  }

  span {
    margin: 0 15px 0 7px;
  }
`

interface Props {
  name: string
  html_url: string
  updated_at: string
  stargazers_count: number
  watchers_count: number
  forks_count: number
}

const ListItem = (props: Props) => {
  const { name, html_url, updated_at, stargazers_count, watchers_count, forks_count } = props

  return (
    <ListItemContainer
      className="list-item"
      onClick={() => {
        window.open(html_url, "_blank")
      }}
    >
      <Title>{name}</Title>

      <ListItemContainerFooter>
        <span>Última atualização: {moment(updated_at).locale("pt-br").format("DD/MM/YYYY")}</span>

        <FooterCounts>
          <CountICon>
            <div className="image-wrapper">
              <Image
                width="16"
                height="16"
                src="/static/images/star-regular.svg"
                alt="Quantos marcaram com uma estrela"
                title="Quantos marcaram com uma estrela"
              />
            </div>
            <span>{stargazers_count}</span>
          </CountICon>
          <CountICon>
            <div className="image-wrapper">
              <Image
                width="16"
                height="16"
                src="/static/images/eye-regular.svg"
                alt="Quantos estão observando"
                title="Quantos estão observando"
              />
            </div>
            <span>{watchers_count}</span>
          </CountICon>
          <CountICon>
            <div className="image-wrapper">
              <Image
                width="16"
                height="16"
                src="/static/images/code-branch-solid.svg"
                alt="Quantos forks foram feitos"
                title="Quantos forks foram feitos"
              />
            </div>
            <span>{forks_count}</span>
          </CountICon>
        </FooterCounts>
      </ListItemContainerFooter>
    </ListItemContainer>
  )
}

export default ListItem
