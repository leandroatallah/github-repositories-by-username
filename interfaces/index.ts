export interface Repository {
  name: string
  html_url: string
  updated_at: string
  stargazers_count: number
  watchers_count: number
  forks_count: number
}

export interface FormContextInterface {
  pageList?: number
  loading?: boolean
  showResult?: boolean
  listRepositories?: Repository[] | null
  errorMessage?: string | null
  publicRepos?: number | null
  setPageList?: (pageList: number) => void
  setLoading?: (loading: boolean) => void
  setShowResult?: (show: boolean) => void
  setListRepositories?: (repositories: any[]) => void
  setErrorMessage?: (error: string | null) => void
  setPublicRepos?: (publicRepos: number | null) => void
}
