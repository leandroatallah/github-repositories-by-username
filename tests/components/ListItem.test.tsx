import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import ListItem from "../../components/ListItem"
import { Repository } from "interfaces"

describe("ListItem", () => {
  it("should render a list item", () => {
    const props: Repository = {
      name: "repo1",
      html_url: "#",
      updated_at: "2018-01-01T00:00:00Z",
      stargazers_count: 1,
      watchers_count: 2,
      forks_count: 3,
    }

    const { container } = render(<ListItem {...props} />)
    expect(container).toMatchSnapshot()
  })
})
