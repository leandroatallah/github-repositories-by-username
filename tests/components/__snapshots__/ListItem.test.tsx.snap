// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render a list item 1`] = `
<div>
  <div
    class="sc-bdnxRM fKvVhg list-item"
  >
    <h2
      class="sc-dlnjwi jcWZPf"
    >
      repo1
    </h2>
    <div
      class="sc-gtsrHT kTrHWf"
    >
      <span>
        Última atualização: 
        12/31/2017
      </span>
      <div
        class="footer-counts"
      >
        <span
          class="sc-hKFxyN kMXQJh"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Quantos marcaram com uma estrela"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              title="Quantos marcaram com uma estrela"
            />
          </div>
          <span>
            1
          </span>
        </span>
        <span
          class="sc-hKFxyN kMXQJh"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Quantos estão observando"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              title="Quantos estão observando"
            />
          </div>
          <span>
            2
          </span>
        </span>
        <span
          class="sc-hKFxyN kMXQJh"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Quantos forks foram feitos"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              title="Quantos forks foram feitos"
            />
          </div>
          <span>
            3
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
